sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -231,6 +325,57 @@ export namespace ConnectionContext {
     return connectionContext;
   }
 
+  /**
+   * Fetches the receiver from the cache in ConnectionContext based on the receiverName given.
+   * Useful for when a message needs to be settled or have its lock renewed.
+   *
+   * TODO: Track the right receiver on the message instead of the ConnectionContext to remove
+   * the need for this helper.
+   *
+   * @internal
+   * @ignore
+   */
+  export function getReceiverFromCache(
+    context: ConnectionContext,
+    receiverName: string,
+    sessionId?: string
+  ): MessageReceiver | MessageSession | undefined {
+    if (sessionId != null && context.messageSessions[receiverName]) {
```

Comment:
Should we also make sure that the messageSession's sessionId matches the sessionId passed in?

Reply:
The receiver name is unique enough, we dont need to check for session.
We can actually remove `sessionId` from this function input altogether.
The one optimization it can serve is that we can skip looking at `context.streamingReceivers` and `context.batchingReceivers` if we know that the given receiver name is associated with sessions.